scripts/generate-docs.go

```diff
@@ -0,0 +1,307 @@
+// This script is designed to precompile all our documentation template files
+// so they are ready to then be passed onto the official terraform
+// documentation generation tool:
+// https://github.com/hashicorp/terraform-plugin-docs
+//
+// The reason why we need to precompile the templates is because they use the
+// {{ define "..." }} syntax to allow us to reuse Markdown content across
+// multiple files, and so we need this script to expose those separate files as
+// if they were defined within a single template file.
+//
+// The reason we need the official terraform documentation generator tool is
+// because it helps to pick up missing content that we've not defined by
+// reflecting over the schemas of our resources and data sources.
+//
+// THE COMPLETE DOCUMENTATION STEPS ARE:
+//
+// 1. acquire all the templates
+// 2. render the templates into Markdown
+// 3. write the file output to a temp directory and still use .tmpl extension
+// 4. append to each rendered .tmpl file the template code needed by tfplugindocs (e.g. {{ .SchemaMarkdown | trimspace }})
+// 5. copy the index.md file (which requires no pre-compiling) to the temporary directory so tfplugindocs can include it
+// 6. rename repo /templates/{data-sources/resources} directories to avoid being overwritten by next step
+// 7. move contents of temp directory (i.e. data-sources/resources) into repo /templates directory
+// 8. run tfplugindocs generate function to output final documentation to /docs directory
+// 9. replace /templates/{data-sources/resources} directories with their backed up equivalents
+//
+package main
+
+import (
+	"fmt"
+	"html/template"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+// PageData represents a type of service and enables a template to render
+// different content depending on the service type.
+type PageData struct {
+	ServiceType string
+}
+
+// Page represents a template page to be rendered.
+//
+// name: the {{ define "..." }} in each template.
+// path: the path to write rendered output to.
+// Data: context specific information (e.g. vcl vs wasm).
+//
+// Data is public as it's called via the template processing logic.
+type Page struct {
+	name string
+	path string
+	Data PageData
+}
+
+func main() {
+	baseDir := getBaseDir()
+	tmplDir := baseDir + "/templates"
+
+	tempDir, err := ioutil.TempDir("", "precompile")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	var dataPages = []Page{
+		{
+			name: "ip_ranges",
+			path: tempDir + "/data-sources/ip_ranges.md.tmpl",
+		},
+		{
+			name: "waf_rules",
+			path: tempDir + "/data-sources/waf_rules.md.tmpl",
+		},
+	}
+
+	var resourcePages = []Page{
+		{
+			name: "service_v1",
+			path: tempDir + "/resources/service_v1.md.tmpl",
+			Data: PageData{
+				"vcl",
+			},
+		},
+		{
+			name: "service_compute",
+			path: tempDir + "/resources/service_compute.md.tmpl",
```

Comment:
I now realise why these files are showing as completely new in the diff for the output directory. The names of these files should match the resource name. I.e.:

```suggestion
			path: tempDir + "/resources/fastly_service_compute.md.tmpl",
```

Similar comments elided for all of these references in this file.

Reply:
This doesn't look to be how it worked originally...

https://github.com/fastly/terraform-provider-fastly/tree/master/website/docs/r

...in there you can see the original 'output' directory (i.e. `website` has the resource file named `service_compute.html.markdown` which is what we have here except we've replaced the extensions with what what tfplugindocs is expecting.)